File: r/src/option.r
Option <- setClass(
    "Option",
    slots = c(
        index="numeric",
        title="character"
    )
)

# Definitions
setGeneric(name="printOption", def=function(self) { standardGeneric("printOption") })

# Implementations
setMethod(f="printOption", signature="Option", definition=function(self) {
    message(self@index, ". ", self@title)
})

File: java/src/com/eliudarudo/cliapp/Option.java
package com.eliudarudo.cliapp;

public class Option {
    private int index;
    private String title;

    public Option(int index, String title) {
        this.index = index;
        this.title = title;
    }

    public void print() {
        System.out.println(this.index + ". " + this.title);
    }
}
